core/domain/activity_jobs_one_off.py

```diff
@@ -698,36 +708,35 @@ def map(snapshot_model):
             if snapshot_model.commit_type in ['create', 'delete']:
                 commit_logs_should_exist = False
 
-        if commit_log_model is None and commit_logs_should_exist:
-            if parent_model is not None:
-                # Public is the default value for post_commit_status for the
-                # question and skill models. For the exp model, it is assigned
-                # from the rights model.
-                commit_log_model = model_properties['commit_log_model_class'](
-                    id=commit_log_id,
-                    user_id=snapshot_model.committer_id,
-                    commit_type=snapshot_model.commit_type,
-                    commit_message=snapshot_model.commit_message,
-                    commit_cmds=snapshot_model.commit_cmds,
-                    post_commit_status=constants.ACTIVITY_STATUS_PUBLIC,
-                    version=int(version)
-                )
-                setattr(
-                    commit_log_model, model_properties['id_field'], model_id)
-                if class_name == 'ExplorationRightsSnapshotMetadataModel':
-                    commit_log_model.post_commit_status = parent_model.status
-                elif class_name == 'ExplorationSnapshotMetadataModel':
-                    rights_model = exp_models.ExplorationRightsModel.get_by_id(
-                        model_id)
-                    commit_log_model.post_commit_status = rights_model.status
-                commit_log_model.put()
-                yield (
-                    'Added missing commit log model-%s' % class_name,
-                    snapshot_model.id)
-            else:
-                yield ('Missing Parent Model-No changes-%s' % class_name, 1)
-        else:
+        if commit_log_model is not None or not commit_logs_should_exist:
             yield ('Found commit log model-%s' % class_name, 1)
+            return
+
+        if parent_model is None:
+            yield ('Missing Parent Model-No changes-%s' % class_name, 1)
+            return
+
+        commit_log_model = model_properties['commit_log_model_class'](
+            id=commit_log_id,
+            user_id=snapshot_model.committer_id,
+            commit_type=snapshot_model.commit_type,
+            commit_message=snapshot_model.commit_message,
+            commit_cmds=snapshot_model.commit_cmds,
+            version=version
+        )
+        setattr(
+            commit_log_model, model_properties['id_field'], model_id)
+        if class_name in job_class.MODELS_NAMES_WITH_DEFAULT_COMMIT_STATUS:
+            commit_log_model.post_commit_status = (
+                constants.ACTIVITY_STATUS_PUBLIC)
+        elif class_name in job_class.MODELS_NAMES_WITH_COMMIT_STATUS_IN_RIGHTS:
+            rights_model = exp_models.ExplorationRightsModel.get_version(
+                model_id, version)
```

Comment:
Careful -- I don't think the version number of the rights model necessarily corresponds to the version number of the exploration model. Can you check whether that's the case? If not, you'll need to do some kind of "binary search by date" instead.

Reply:
I have made a query for the correct version. Done.

Reply:
Since the exploration snapshot models is not being queried, there is no need to query for the correct version